### PR TITLE
Added NetStandard2.0 target framework to reduce dependencies for NetCore

### PR DESCRIPTION
**Pull request by snakefoot**

### CODE REVIEW

**Comment by guptakeshav-sumo** [comment]
We are already having `*.netstandard.csproj` files targetting `netstandard2.0` corresponding to each `*.csproj`. Do we still need these changes?

**Comment by snakefoot** [comment]
Don't know what that is. But right now the nuget packages doesnt include netstandard2. 0


Go here and expand dependencies and you will only see net45 and legacy netstandard1.3


https://www.nuget.org/packages/SumoLogic.Logging.Common/


This PR will fix this by adding netstandard2.0 which reduces dependencies for NeCoreApp2+3




Sent from my Sony Xperia


---- Keshav Gupta wrote ----


We are already having *.netstandard.csproj files targetting netstandard2.0 corresponding to each *.csproj. Do we still need these changes?

**Comment by snakefoot** [comment]
Ahh you mean these:

https://github.com/SumoLogic/sumologic-net-appenders/blob/master/SumoLogic.Logging.Common/SumoLogic.Logging.Common.netstandard.csproj

And this:

https://github.com/SumoLogic/sumologic-net-appenders/blob/master/SumoLogic.Logging.netstandard.sln

Think all of those are legacy from before multi-targetframeworks was known:

https://github.com/SumoLogic/sumologic-net-appenders/blob/073ea057bf4a945e5e7b66eecc6e371c4b49c264/SumoLogic.Logging.Common/SumoLogic.Logging.Common.csproj#L7

You can probably remove the `*.netstandard.csproj`-files after having merged this PR. Unless you are using them yourself somewhere internally.

**Comment by snakefoot** [comment]
Looking at your appveyor-build-script:

https://github.com/SumoLogic/sumologic-net-appenders/blob/master/ci/build-solution.ps1

Then it has special condition when running on Linux. But by default it builds the normal packages.

I can make a new PR that removes all this "NetStandard" noise, when this PR has been merged.

File: SumoLogic.Logging.Log4Net/SumoLogic.Logging.Log4Net.csproj [hunk 1]
```diff
@@ -4,10 +4,10 @@
         <Description>Log4Net appender which sends logs to the Sumo Logic machine data analytics platform.</Description>
         <Authors>Sumo Logic</Authors>
         <Copyright>Copyright © 2018 Sumo Logic Inc. - All Rights Reserved</Copyright>
-        <TargetFrameworks>netstandard1.3;net45</TargetFrameworks>
+        <TargetFrameworks>net45;netstandard1.3</TargetFrameworks>
```
Comment (guptakeshav-sumo) [inline SumoLogic.Logging.Log4Net/SumoLogic.Logging.Log4Net.csproj]: `netstandard2.0` here?

File: SumoLogic.Logging.Common/SumoLogic.Logging.Common.csproj [hunk 2]
```diff
@@ -16,20 +16,15 @@
...
-        <PackageReference Include="System.Net.Requests" Version="4.3.0" />
+      <None Include="..\SumoLogic.Logging.snk">
+          <Link>SumoLogic.Logging.snk</Link>
+      </None>
+      <CodeAnalysisDictionary Include="..\CustomDictionary.xml">
+          <Link>CustomDictionary.xml</Link>
+      </CodeAnalysisDictionary>
+    </ItemGroup>
+    <ItemGroup Condition=" '$(TargetFramework)' != 'netstandard2.0' ">
+      <PackageReference Include="System.Net.Http" Version="4.3.1" />
```
Comment (guptakeshav-sumo) [inline SumoLogic.Logging.Common/SumoLogic.Logging.Common.csproj]: Tab spacing seems to have changed - can we please revert back?

File: SumoLogic.Logging.Common/Sender/SumoLogicMessageSender.cs [hunk 3]
```diff
@@ -161,7 +161,11 @@ public async Task Send(string body, string name, string category, string host)
                 }
                 catch (Exception ex)
                 {
-                    if (!(ex is IOException || ex is HttpRequestException || ex is WebException))
+#if NETSTANDARD
```
Comment (guptakeshav-sumo) [inline SumoLogic.Logging.Common/Sender/SumoLogicMessageSender.cs]: From where is the constant coming from?

**Comment by snakefoot** [comment]
Log4net does not have a netstandard2.0 target. So no need.


Have not changed tab/spaces it is just comparison getting confused and not working well with xml.


NETSTANDARD comes from the Microsoft build tools. Just like NET45 exist for Net45 build.


Sent from my Sony Xperia


---- Keshav Gupta wrote ----


We are already having *.netstandard.csproj files targetting netstandard2.0 corresponding to each *.csproj. Do we still need these changes?

File: SumoLogic.Logging.Log4Net/SumoLogic.Logging.Log4Net.csproj (hunk 1 above)
Comment (snakefoot) [reply SumoLogic.Logging.Log4Net/SumoLogic.Logging.Log4Net.csproj]: Log4net does not have a netstandard2.0 target. So no need.

File: SumoLogic.Logging.Common/Sender/SumoLogicMessageSender.cs (hunk 3 above)
Comment (snakefoot) [reply SumoLogic.Logging.Common/Sender/SumoLogicMessageSender.cs]: NETSTANDARD comes from the Microsoft build tools. Just like NET45 exist for Net45 build.

File: SumoLogic.Logging.Common/SumoLogic.Logging.Common.csproj (hunk 2 above)
Comment (snakefoot) [reply SumoLogic.Logging.Common/SumoLogic.Logging.Common.csproj]: Have not changed tab/spaces it is just github-comparison getting confused and not working well with xml. VS2019-comparison looks correct.

**Comment by snakefoot** [comment]
Created  #96 as followup to removing `*.netstandard.csproj`-files

**Pull request merged**